repository.sls

```diff
@@ -1,14 +1,26 @@
 # Setup official docker repository
-apt-transport-https:
-  pkg.installed: []
+{% if grains['osrelease']|float < 22.04  %}
```

Comment:
I don't think it's necessary, as apt-key is [supported since Ubuntu 16.04](https://askubuntu.com/a/1392143). It was merely not enforced until now.

Reply:
We had problems in the past with older Ubuntu versions, that's the reason for the version check. We are using this approach already for all other formulas, too. So it's also consistent to our other formulas.

Reply:
Mmh, the formulas here are not using this? See https://github.com/chr4/salt-postgresql/blob/main/repository.sls

If it works for 18.04 and 20.04 that would be fine for me - that should be the case, shouldn't it?

Reply:
Nope, as I mentioned, we had problems with Ubuntu 20.04. We would have to make sure, that the directory for the keys is created with the right permissions, as it doesn't exist on a Ubuntu 20.04 machine.

```
man@1:~$ cat /etc/issue.net
Ubuntu 20.04.6 LTS
man@1:~$ ls -l /etc/apt/keyrings
ls: cannot access '/etc/apt/keyrings': No such file or directory
man@1:~$
```

So from my point of view, it's not really an advantage to manage this directory manually compared to the suggested approach (which we use in 8 different places already without any issues).

Reply:
I've just tested the linked psql repo - it fails silently on an Ubuntu 20.04 machine. So we have to update this as well, but I just wanted to make sure, that we agree on the implementation, before opening multiple MRs.

Reply:
Good to know! Doesn't speak for the postgresql integration tests 🙈 

I suppose both ways would be fine. Using seperate keyrings has a security benefit, but there's another possible catch: The directory is part of the `apt` package on 22.04, so I'm not sure about the exact behavior of `do-release-upgrade` if the directory was manually created. 

Correcy permissions would be `0755` and `root:root` - but if you tested this I think we can keep it, I'm convinced. :)